### PR TITLE
Parse records

### PR DESCRIPTION
**Pull request by adamdickmeiss**
This is just the first step of parsing the request from OCLC Connexion client. The next PR will be calling mod-copycat and me trying to figure out how the edge-common setup really works!

### CODE REVIEW

File: src/main/java/org/folio/edge/connexion/MainVerticle.java [hunk 1]
```diff
@@ -11,55 +11,67 @@
...
-                    }).listen(port).mapEmpty();
-        }).onComplete(promise);
-    }
+  @Override
+  public void start(Promise<Void> promise) {
+    Future.<Void>future(p -> super.start(p)).<Void>compose(res -> {
+      port = config().getInteger(SYS_PORT, DEFAULT_PORT);
+      return vertx.createNetServer()
+          .connectHandler(socket -> {
+            boolean [] handled = new boolean[] { false };
```
Comment (skoczko) [inline src/main/java/org/folio/edge/connexion/MainVerticle.java:34]: Is this a trick to easilly "box" primitives?

Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/connexion/MainVerticle.java:34]: Yes. Needs to be final

Comment (julianladisch) [reply src/main/java/org/folio/edge/connexion/MainVerticle.java:34]: Another way is to use an https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/atomic/AtomicBoolean.html

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java [hunk 2]
```diff
@@ -48,4 +49,35 @@ public void testAdminHealth(TestContext context) {
...
   }
+
+  @Test
+  public void testImportWithUserOK(TestContext context) {
+    final int port = 9230;
+    JsonObject config = new JsonObject();
+    config.put(SYS_PORT, port);
+
+    NetClient netClient = vertx.createNetClient();
+    netClient.connect(port, "localhost").onComplete(context.asyncAssertFailure(f ->
```
Comment (julianladisch) [inline src/test/java/org/folio/edge/connexion/MainVerticleTest.java:60]: There is no need to check that the port is free. deployVerticle will fail if the port is in use.

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java [hunk 3]
```diff
@@ -48,4 +49,35 @@ public void testAdminHealth(TestContext context) {
...
+        vertx.deployVerticle(new MainVerticle(), new DeploymentOptions().setConfig(config))
```
Comment (julianladisch) [inline src/test/java/org/folio/edge/connexion/MainVerticleTest.java:61]: The duplicate code should be moved into a method:

    private Future<Void> deploy(MainVerticle mainVerticle) {
      final int port = 9230;
      JsonObject config = new JsonObject();
      config.put(SYS_PORT, port);
      return vertx.deployVerticle(mainVerticle, new DeploymentOptions().setConfig(config));
    }

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java [hunk 4]
```diff
@@ -48,4 +49,35 @@ public void testAdminHealth(TestContext context) {
...
+            .compose(x -> netClient.connect(port, "localhost").compose(socket -> socket.write("U6MyUser")))
```
Comment (julianladisch) [inline src/test/java/org/folio/edge/connexion/MainVerticleTest.java:62]:
```suggestion
            .compose(x -> netClient.connect(port, "localhost"))
            .compose(socket -> socket.write("U6MyUser"))
```

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/edge/connexion/MainVerticleTest.java:60]: sure

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/edge/connexion/MainVerticleTest.java:61]: OK.. Done that now, but it deliberately waited with this as there will be changers to config and or MainVerticle..

File: src/main/java/org/folio/edge/connexion/MainVerticle.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/connexion/MainVerticle.java:34]: Sure.. It's another alternative, but a complex one for something so simple (eg thread safete and all for something so simple)

**Comment by adamdickmeiss** [comment]
Thx for taking a look @julianladisch

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java [hunk 5]
```diff
@@ -35,17 +40,41 @@ public void instanceMainVerticvle() {
...
+        .onComplete(context.asyncAssertSuccess());
+  }
+
+  @Test
+  public void testImportWithUserOK(TestContext context) {
+    NetClient netClient = vertx.createNetClient();
+    deploy(new MainVerticle())
+        .compose(x -> netClient.connect(PORT, "localhost")
+            .compose(socket -> socket.write("U6MyUser")))
+        .onComplete(context.asyncAssertSuccess());
```
Comment (julianladisch) [inline src/test/java/org/folio/edge/connexion/MainVerticleTest.java:64]:
```suggestion
        .compose(x -> netClient.connect(PORT, "localhost"))
        .compose(socket -> socket.write("U6MyUser"))
        .onComplete(context.asyncAssertSuccess());
```

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java [hunk 6]
```diff
@@ -35,17 +40,41 @@ public void instanceMainVerticvle() {
...
+    deploy(new MainVerticle())
+        .compose(x -> webClient.get(PORT, "localhost", "/admin/health").send())
+        .onComplete(context.asyncAssertSuccess());
+  }
+
+  @Test
+  public void testImportWithUserOK(TestContext context) {
+    NetClient netClient = vertx.createNetClient();
+    deploy(new MainVerticle())
+        .compose(x -> netClient.connect(PORT, "localhost")
```
Comment (julianladisch) [inline src/test/java/org/folio/edge/connexion/MainVerticleTest.java:62]:
```suggestion
        .compose(x -> vertx.createNetClient().connect(PORT, "localhost")
```
The netClient variable is no longer needed.

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java [hunk 7]
```diff
@@ -35,17 +40,41 @@ public void instanceMainVerticvle() {
...
+  public void testImportTooLargeMessage(TestContext context) {
+    MainVerticle mainVerticle = new MainVerticle();
+    mainVerticle.setMaxRecordSize(10);
+    NetClient netClient = vertx.createNetClient();
+    netClient.connect(PORT, "localhost").onComplete(context.asyncAssertFailure(f ->
+        deploy(mainVerticle)
+            .compose(x -> netClient.connect(PORT, "localhost")
+                .compose(socket -> socket.write("00006123456")))
             .onComplete(context.asyncAssertSuccess())
     ));
```
Comment (julianladisch) [inline src/test/java/org/folio/edge/connexion/MainVerticleTest.java:77]:
```suggestion
    deploy(mainVerticle)
        .compose(x -> vertx.createNetClient().connect(PORT, "localhost"))
        .compose(socket -> socket.write("00006123456"))
        .onComplete(context.asyncAssertSuccess());
```
The same cleanup as in testImportWithUserOK.

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java [hunk 8]
```diff
@@ -16,6 +18,9 @@
 
 @RunWith(VertxUnitRunner.class)
 public class MainVerticleTest {
+  // RMB has a utility to find a free port, and edge-common has its own testing..
+  // we will use the same port as in Okapi does it its testing mode.
+  private final int PORT = 9230;
 
   Vertx vertx;
```
Comment (julianladisch) [inline src/test/java/org/folio/edge/connexion/MainVerticleTest.java:25]: This is just a remark, not a change request:
In vertx-junit5 there is no need to create and close Vertx in before() and after() methods. vertx-junit5 automatically creates and closes a Vertx for each test method:
* https://vertx.io/docs/vertx-junit5/java/#_test_methods
* https://vertx.io/docs/vertx-junit5/java/#_scope_of_vertx_objects
* https://vertx.io/docs/vertx-junit5/java/#_closing_and_removal_of_vertx_objects

File: src/test/java/org/folio/edge/connexion/MainVerticleTest.java (hunk 7 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/edge/connexion/MainVerticleTest.java]: ok

**Review by skoczko — APPROVED** [review]

**Pull request merged**